### ISSUE TITLE
ORC-1531: Create `orc-format` module and repo

### ISSUE THREAD

**Issue by dongjoon-hyun** [apache/orc#1543]
- https://github.com/apache/orc-format
- https://repository.apache.org/content/groups/snapshots/org/apache/orc/orc-format/1.0.0-SNAPSHOT/

**Comment by deshanxiao** [comment]
Thank you for adding the task. @dongjoon-hyun

**Comment by mystic-lama** [comment]
Is the ask to create a repo very similar to [parquet-format](https://github.com/apache/parquet-format), that shall host the specification and [orc_proto.proto](https://github.com/mystic-lama/orc/blob/main/proto/orc_proto.proto). Possibly generating bindings too.

**Comment by dongjoon-hyun** [comment]
Yes, that's the model, @mystic-lama .

**Comment by mystic-lama** [comment]
Did a quick take at this and created the [orc-format](https://github.com/mystic-lama/orc-format-init) repo. If this is going in right direction, can continue to work on it. Feedback appreaciated

**Comment by mystic-lama** [comment]
Create JIRA [ORC-1531](https://issues.apache.org/jira/browse/ORC-1531) to track the work and discussion

**Comment by dongjoon-hyun** [comment]
Thank you, @mystic-lama . I took a look. 

For the versioning,

https://github.com/mystic-lama/orc-format-init/blob/1f88cb6cc9fd47f430e78aa212923635550fd710/pom.xml#L25C7-L25C7

The following is a little too much for me because we are still on `ORCv1` and will not introduce any format-wise change.
```
2.0.0-SNAPSHOT
```

**Comment by dongjoon-hyun** [comment]
In addition, we don't need `nohive` and `shaded-protobuf` classifier for now.

```
$ ls -al
total 9176
drwxr-xr-x  8 dongjoon  staff      256 Dec  3 15:15 .
drwxr-xr-x  4 dongjoon  staff      128 Dec  3 15:15 ..
-rw-r--r--  1 dongjoon  staff      291 Dec  3 15:15 _remote.repositories
-rw-r--r--  1 dongjoon  staff     1127 Dec  3 15:15 maven-metadata-local.xml
-rw-r--r--  1 dongjoon  staff  2109804 Dec  3 15:15 orc-format-2.0.0-SNAPSHOT-nohive.jar
-rw-r--r--  1 dongjoon  staff  2109804 Dec  3 15:15 orc-format-2.0.0-SNAPSHOT-shaded-protobuf.jar
-rw-r--r--  1 dongjoon  staff   449522 Dec  3 15:15 orc-format-2.0.0-SNAPSHOT.jar
-rw-r--r--  1 dongjoon  staff     8240 Dec  3 15:15 orc-format-2.0.0-SNAPSHOT.pom
```

**Comment by dongjoon-hyun** [comment]
I removed the shading from @mystic-lama 's repo like the following.
- https://github.com/dongjoon-hyun/orc/commit/60c33ac67d8eedbef3324d43e911e88f1f434bc8

```
$ git diff
diff --git a/pom.xml b/pom.xml
index 5d9b72e..32cbd04 100644
--- a/pom.xml
+++ b/pom.xml
@@ -143,92 +143,6 @@
                     <destDir>${project.artifactId}</destDir>
                 </configuration>
             </plugin>
-            <plugin>
-                <groupId>org.apache.maven.plugins</groupId>
-                <artifactId>maven-shade-plugin</artifactId>
-                <version>${maven-shade-plugin.version}</version>
-                <executions>
-                    <execution>
-                        <id>shaded-protobuf</id>
-                        <goals>
-                            <goal>shade</goal>
-                        </goals>
-                        <phase>package</phase>
-                        <configuration>
-                            <artifactSet>
-                                <includes>
-                                    <include>com.google.protobuf:protobuf-java</include>
-                                </includes>
-                            </artifactSet>
-                            <shadedArtifactAttached>true</shadedArtifactAttached>
-                            <shadedClassifierName>shaded-protobuf</shadedClassifierName>
-                            <relocations>
-                                <relocation>
-                                    <pattern>com.google.protobuf</pattern>
-                                    <shadedPattern>org.apache.orc.protobuf</shadedPattern>
-                                </relocation>
-                            </relocations>
-                            <filters>
-                                <filter>
-                                    <artifact>*:*</artifact>
-                                    <excludes>
-                                        <exclude>module-info.class</exclude>
-                                        <exclude>META-INF/MANIFEST.MF</exclude>
-                                        <exclude>META-INF/DEPENDENCIES</exclude>
-                                        <exclude>META-INF/LICENSE</exclude>
-                                        <exclude>META-INF/NOTICE</exclude>
-                                        <exclude>google/protobuf/**</exclude>
-                                    </excludes>
-                                </filter>
-                            </filters>
-                        </configuration>
-                    </execution>
-                    <execution>
-                        <id>nohive</id>
-                        <goals>
-                            <goal>shade</goal>
-                        </goals>
-                        <phase>package</phase>
-                        <configuration>
-                            <artifactSet>
-                                <includes>
-                                    <include>com.google.protobuf:protobuf-java</include>
-                                    <include>org.apache.hive:hive-storage-api</include>
-                                </includes>
-                            </artifactSet>
-                            <shadedArtifactAttached>true</shadedArtifactAttached>
-                            <shadedClassifierName>nohive</shadedClassifierName>
-                            <relocations>
-                                <relocation>
-                                    <pattern>com.google.protobuf</pattern>
-                                    <shadedPattern>org.apache.orc.protobuf</shadedPattern>
-                                </relocation>
-                                <relocation>
-                                    <pattern>org.apache.hadoop.hive</pattern>
-                                    <shadedPattern>org.apache.orc.storage</shadedPattern>
-                                </relocation>
-                                <relocation>
-                                    <pattern>org.apache.hive</pattern>
-                                    <shadedPattern>org.apache.orc.storage</shadedPattern>
-                                </relocation>
-                            </relocations>
-                            <filters>
-                                <filter>
-                                    <artifact>*:*</artifact>
-                                    <excludes>
-                                        <exclude>module-info.class</exclude>
-                                        <exclude>META-INF/MANIFEST.MF</exclude>
-                                        <exclude>META-INF/DEPENDENCIES</exclude>
-                                        <exclude>META-INF/LICENSE</exclude>
-                                        <exclude>META-INF/NOTICE</exclude>
-                                        <exclude>google/protobuf/**</exclude>
-                                    </excludes>
-                                </filter>
-                            </filters>
-                        </configuration>
-                    </execution>
-                </executions>
-            </plugin>
```

**Comment by dongjoon-hyun** [comment]
1. I sent an email to the dev mailing list.
  - https://lists.apache.org/thread/4kx0bxp34lnggsx6rhyx23t68vf4hqoj
2. In a few days, I'll file an ASF JIRA issue to create a new repo.
  - You can make the initial PR to the new repo and will start to vote for the Apache `orc-format` 1.0.0 release.

**Comment by deshanxiao** [comment]
Hi @dongjoon-hyun It is time to reconsider this proposal? #1516

**Comment by mystic-lama** [comment]
Thanks @dongjoon-hyun I shall address the comments and we can do a final refinement during the PR.

Thanks a lot for looking into this.

**Comment by wgtmac** [comment]
> @wgtmac I thought about this part, but I am not very well versed on the C++ side of things. Here I can use your expertise on moving forward

My proposal is to simply add apache/orc-format as a git submodule to apache/orc-cpp repo. Either apache/orc-format prepares generated cpp source files directly, or apache/orc-cpp compiles the proto file from it. WDYT?

cc @stiga-huang @coderex2522 @ffacs

**Comment by mystic-lama** [comment]
Got it. We can collect feedback during PR in case we have to adjust thing around for easier integration.

**Comment by dongjoon-hyun** [comment]
Sorry, but actually, I'm negative about a new `orc-cpp` repo at this time, @wgtmac .
> apache/orc-cpp

**Comment by wgtmac** [comment]
That makes sense. A single repo makes things a lot easier. @dongjoon-hyun

**Comment by dongjoon-hyun** [comment]
Thank you, @wgtmac .

**Comment by dongjoon-hyun** [comment]
To all, I created `orc-format` repo here. Currently, it's empty.
- https://github.com/apache/orc-format

**Comment by mystic-lama** [comment]
@dongjoon-hyun Thank you so much.

I am not able to fork it to send PR as it is empty, Suggestions?

**Comment by dongjoon-hyun** [comment]
Oh, it might be because it's empty. Let me bring your change with your authorship.

**Comment by dongjoon-hyun** [comment]
BTW, @mystic-lama . It would be great if you can use your public real name. Actually, that's ASF policy for new JIRA accounts. Eventually, you will need it.
```
mystic-lama <mysticlama000@gmail.com>
```

**Comment by dongjoon-hyun** [comment]
I commit your first commit. Could you retry to clone?
- https://github.com/apache/orc-format/commit/37c4581335c73f72f92329bf4d4f3c8cd60c0369

**Comment by dongjoon-hyun** [comment]
FYI, I filed INFRA jira to publish the snapshot of `orc-format`.
- [INFRA-25244 Setup NEXUS_USER and NEXUS_PW secrets at `orc-format`](https://issues.apache.org/jira/browse/INFRA-25244)

**Comment by mystic-lama** [comment]
> * https://github.com/apache/orc-format

It works. Have archived the poc repository

**Comment by dongjoon-hyun** [comment]
ASF INFRA team resolved the issue and Apache ORC Format 1.0.0-SNAPSHOT publishing is started.
- https://github.com/apache/orc-format/actions/runs/7112527742/job/19362640572
- https://repository.apache.org/content/groups/snapshots/org/apache/orc/orc-format/1.0.0-SNAPSHOT/

During the testing with Apache Spark, I found that I mislead you about sharing, @mystic-lama . The shading is added back.

**Comment by dongjoon-hyun** [comment]
Let me close this issue.
